packages/scaffold-ui/src/modal/w3m-modal/index.ts

```diff
@@ -249,13 +249,9 @@ export class W3mModal extends LitElement {
     const authenticated = AccountController.state.siweStatus === 'success'
 
     if (!authenticated && isEIP155Namespace) {
-      if (this.open) {
-        RouterController.replace('ConnectingSiwe')
-      } else {
-        ModalController.open({
-          view: 'ConnectingSiwe'
-        })
-      }
+      ModalController.open({
```

Comment:
One thing is ModalController.open has `await ApiController.state.prefetchPromise` logic. This might cause us to refetch many things unnecessarily.